### PR TITLE
Code style fixes

### PR DESCRIPTION
**Pull request by heuermh**
Instead of finishing up docs on the plane last night, I ran through and made some code style fixes.  Some may overlap with those in #1278.

### CODE REVIEW

**Comment by ryan-williams** [comment]
nice! were you able to do these automatically somehow?

what's the advantage of `final`ing the things that you've done here?

**Comment by heuermh** [comment]
> nice! were you able to do these automatically somehow?

Nope, a three hour flight with no kids helps.  :)

> what's the advantage of finaling the things that you've done here?

Good Java code style, see e.g. Effective Java "Item 17: Design and document for inheritance or else prohibit it" or the Checkstyle [DesignForExtension check](http://checkstyle.sourceforge.net/config_design.html#DesignForExtension).

**Comment by fnothaft** [comment]
>> what's the advantage of finaling the things that you've done here?
>
> Good Java code style, see e.g. Effective Java "Item 17: Design and document for inheritance or else prohibit it" or the Checkstyle DesignForExtension check.

+1, akin to conservatively making methods private/protected.

File: adam-apis/src/main/scala/org/bdgenomics/adam/apis/java/JavaADAMContext.scala [hunk 1]
```diff
@@ -31,16 +28,14 @@ import org.bdgenomics.adam.rdd.variant.{
...
-import scala.collection.JavaConversions._
 
 object JavaADAMContext {
   // convert to and from java/scala implementations
   implicit def fromADAMContext(ac: ADAMContext): JavaADAMContext = new JavaADAMContext(ac)
   implicit def toADAMContext(jac: JavaADAMContext): ADAMContext = jac.ac
 }
 
-class JavaADAMContext private (val ac: ADAMContext) extends Serializable {
+class JavaADAMContext(val ac: ADAMContext) extends Serializable {
```
Comment (fnothaft) [inline adam-apis/src/main/scala/org/bdgenomics/adam/apis/java/JavaADAMContext.scala]: While we're making this public, we should probably remove the implicits in the companion object, no? They'll be unusable from Javaland.

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/AboutSuite.scala [hunk 2]
```diff
@@ -48,6 +48,6 @@ class AboutSuite extends FunSuite {
     assert(!about.scalaVersion().isEmpty)
 
     assert(about.version() !== null)
-    assert(!about.version().isEmpty())
+    assert(!about.version().isEmpty)
```
Comment (fnothaft) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/AboutSuite.scala]: Might as well swap to `nonEmpty` here too.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala [hunk 3]
```diff
@@ -107,7 +106,7 @@ private[adam] class SAMRecordConverter extends Serializable with Logging {
         // set read alignment flag
         val start: Int = samRecord.getAlignmentStart
         assert(start != 0, "Start cannot equal 0 if contig is set.")
-        builder.setStart((start - 1))
+        builder.setStart((start - 1L))
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala]: Remove extra parens here as well?

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 4]
```diff
@@ -149,8 +144,6 @@ private class FileFilter(private val name: String) extends PathFilter {
   }
 }
 
-import org.bdgenomics.adam.rdd.ADAMContext._
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Nice! I was hoping we could get rid of this.

(On reflection, I don't know why I didn't just try deleting it in the first place!)

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2ADAM.scala [hunk 5]
```diff
@@ -60,8 +56,8 @@ class Vcf2ADAM(val args: Vcf2ADAMArgs) extends BDGSparkCommand[Vcf2ADAMArgs] wit
   def run(sc: SparkContext) {
 
     val variantContextRdd = sc.loadVcf(args.vcfPath)
-    var variantContextsToSave = if (args.coalesce > 0) {
-      if (args.coalesce > variantContextRdd.rdd.partitions.size || args.forceShuffle) {
+    val variantContextsToSave = if (args.coalesce > 0) {
+      if (args.coalesce > variantContextRdd.rdd.partitions.length || args.forceShuffle) {
         variantContextRdd.transform(_.coalesce(args.coalesce, shuffle = true))
```
Comment (devin-petersohn) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2ADAM.scala]: We can cut this block down to one line and don't have to do the if then else:
variantContextRdd.transform(_.coalesce(args.coalesce, shuffle = (args.coalesce > variantContextRdd.rdd.partitions.length || args.forceShuffle)))

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2ADAM.scala [hunk 6]
```diff
@@ -60,8 +56,8 @@ class Vcf2ADAM(val args: Vcf2ADAMArgs) extends BDGSparkCommand[Vcf2ADAMArgs] wit
   def run(sc: SparkContext) {
 
     val variantContextRdd = sc.loadVcf(args.vcfPath)
-    var variantContextsToSave = if (args.coalesce > 0) {
-      if (args.coalesce > variantContextRdd.rdd.partitions.size || args.forceShuffle) {
+    val variantContextsToSave = if (args.coalesce > 0) {
+      if (args.coalesce > variantContextRdd.rdd.partitions.length || args.forceShuffle) {
```
Comment (devin-petersohn) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2ADAM.scala]: We can cut this if then else out:
`variantContextRdd.transform(_.coalesce(args.coalesce, shuffle = (args.coalesce > variantContextRdd.rdd.partitions.length || args.forceShuffle)))`

**Comment by heuermh** [comment]
Addressed review comments. It might be more clear in git history to implement the change(s) in #1298 and then merge this pull request after that one.

File: adam-apis/src/main/scala/org/bdgenomics/adam/apis/java/JavaADAMContext.scala (hunk 1 above)
Comment (fnothaft) [inline adam-apis/src/main/scala/org/bdgenomics/adam/apis/java/JavaADAMContext.scala]: I've just merged #1298. Want to rebase on that and reprise this point?

**Comment by heuermh** [comment]
Rebased and squashed commits after #1298 was merged.  Ready to go as far as I can see.

**Comment by fnothaft** [comment]
Merged! Thanks @heuermh!

**Comment by heuermh** [comment]
Thank you!

**Pull request merged**